osu.Game/Screens/Backgrounds/BackgroundScreenDefault.cs

```diff
@@ -22,17 +23,20 @@ public class BackgroundScreenDefault : BackgroundScreen
 
         private string backgroundName => $@"Menu/menu-background-{currentDisplay % background_count + 1}";
 
-        private Bindable<User> user;
         private Bindable<Skin> skin;
+        private readonly Bindable<WorkingBeatmap> beatmap = new Bindable<WorkingBeatmap>();
+        private readonly Bindable<MainMenuBackgroundMode> backgroundMode = new Bindable<MainMenuBackgroundMode>();
 
         [BackgroundDependencyLoader]
-        private void load(IAPIProvider api, SkinManager skinManager)
+        private void load(SkinManager skinManager, OsuConfigManager config, Bindable<WorkingBeatmap> workingBeatmap)
         {
-            user = api.LocalUser.GetBoundCopy();
```

Comment:
why is this removed? no test coverage to show that this is a regression is a bit of a worry.

Reply:
soo should I just move it back? But we already have supporter check in the settings

Reply:
What happens if you logout or change user?

Reply:
It will change the background mode to default and block the dropdown

Reply:
What if the settings overlay isn't loaded? There's no guarantee it will be.

Also I'm pretty sure that logic you have will crash if you go from non-supporter to non-supporter (it will try and change the bindable value while already in a disabled state).

Yes, that logic should probably stay there, but it should not reset the setting nor should it be responsible for governing the setting.

Reply:
Yeah, looks/feels better. And there's no setting hard reset anymore